test/e2e/tests/confirmations/signatures/permit.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { strict as assert } from 'assert';
+import { Suite } from 'mocha';
+import { withRedesignConfirmationFixtures } from '../helper-fixture';
+import {
+  DAPP_HOST_ADDRESS,
+  WINDOW_TITLES,
+  openDapp,
+  unlockWallet,
+} from '../../../helpers';
+import { Ganache } from '../../../seeder/ganache';
+import { Driver } from '../../../webdriver/driver';
+
+describe('Confirmation Signature - Permit', function (this: Suite) {
+  if (!process.env.ENABLE_CONFIRMATION_REDESIGN) { return; }
+
+  it('initiates and confirms', async function () {
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
+      async ({ driver, ganacheServer }: { driver: Driver, ganacheServer: Ganache }) => {
+        const addresses = await ganacheServer.getAccounts();
+        const publicAddress = addresses?.[0] as string;
+
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await driver.clickElement('#signPermit');
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        const origin = driver.findElement({ text: DAPP_HOST_ADDRESS });
+        const contractPetName = driver.findElement({
+          css: '.name__value',
+          text: '0xCcCCc...ccccC',
+        });
+
+        const primaryType = driver.findElement({ text: 'Permit' });
+        const owner = driver.findElement({ css: '.name__name', text: 'Account 1' });
+        const spender = driver.findElement({ css: '.name__value', text: '0x5B38D...eddC4' });
+        const value = driver.findElement({ text: '3000' });
+        const nonce = driver.findElement({ text: '0' });
+        const deadline = driver.findElement({ text: '50000000000' });
+
+        assert.ok(await origin, 'origin');
+        assert.ok(await contractPetName, 'contractPetName');
+
+        assert.ok(await primaryType, 'primaryType');
+        assert.ok(await owner, 'owner');
+        assert.ok(await spender, 'spender');
+        assert.ok(await value, 'value');
+        assert.ok(await nonce, 'nonce');
+        assert.ok(await deadline, 'deadline');
```

Comment:
Perhaps we can extract this to another assert method?

Reply:
Possibly. Is this to clean the function and have shorter function lengths, or a different specific asset method? cc: @pedronfigueiredo

Reply:
I'm not sure what you mean by asset method in this context, but this is what I had in mind

```js
async function assertSignatureElementsRendered(driver) {
        const origin = driver.findElement({ text: DAPP_HOST_ADDRESS });
        assert.ok(await origin, 'origin');

        const contractPetName = driver.findElement({
            css: '.name__value',
            text: '0xCcCCc...ccccC',
        });
        assert.ok(await contractPetName, 'contractPetName');

        const primaryType = driver.findElement({ text: 'Permit' });
        assert.ok(await primaryType, 'primaryType');

        const owner = driver.findElement({ css: '.name__name', text: 'Account 1' });
        assert.ok(await owner, 'owner');

        const spender = driver.findElement({ css: '.name__value', text: '0x5B38D...eddC4' });
        assert.ok(await spender, 'spender');

        const value = driver.findElement({ text: '3000' });
        assert.ok(await value, 'value');

        const nonce = driver.findElement({ text: '0' });
        assert.ok(await nonce, 'nonce');

        const deadline = driver.findElement({ text: '50000000000' });
        assert.ok(await deadline, 'deadline');
}
```

Reply:
I see. Okay can do 👍🏼

Reply:
Interesting, I wasn't aware of that, but I wonder if there's enough of a difference to justify prioritizing that over optimal code readability/organization

Reply:
This is a nit, feel free to ignore that part of the suggestion if it makes more sense to you

Reply:
actually, you are right. performance is negligible in this case. time 24 → time 25. I believe I realized this while testing elements that rendered with a delay in the test-dapp.  

In regards to readability, I don't quite agree yet the way you wrote it is more readable. I also wrote the code with the same concern for readability in mind. I found it easier skim the function when the method names aligned. The minimal delta in line lengths in consecutive order aids as well. With this, we also can understand what we are asserting by reading half of the lines rather than all lines of the function. I agree it comes with a cost of separating the const with the use in assert. Happy to change this if you or others still feel strongly about the update
<img width="747" alt="Screenshot 2024-05-07 at 7 06 46 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/6e061d20-5e46-47c1-9e41-b1c1ff406d9f">